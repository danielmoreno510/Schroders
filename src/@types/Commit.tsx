/* eslint-disable */
interface ICommit {
  id: number;
  node_id: string;
  title: string;
  user: {
    login: string;
    id: string;
    avatar_url: string;
  };
  comments: number;
  created_at: string;
  body: string;
}

interface ICommitListProps {
  isFetchingCommitList: boolean;
  commitList: ICommit[];
  getCommits: () => void;
  setCurrentCommit(commit: ICommit): any;
  navigation: Navigate;
}

interface ICommitInitialState {
  isFetchingCommitList?: boolean;
  commitList?: ICommit[];
  isFetchingCommitDetails?: boolean;
  commit?: ICommit | null;
  year?: string;
}

interface ICommitReducer {
  type: string;
  commit?: ICommit;
  commitList?: ICommit[];
  year?: string;
}

interface ICommitListStateProps {
  commits: ICommitInitialState;
}

interface ICommitListResponse {
  data: ICommitListResponseData;
}

interface ICommitListResponseData {
  total_count: number;
  incomplete_results: boolean;
  items: ICommit[];
}

interface CommitProps {
  commit: ICommit;
  showCommit: (commit: ICommit) => void;
}
