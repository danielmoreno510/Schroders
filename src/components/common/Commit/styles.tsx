import {StyleSheet} from 'react-native';

import {primaryColor} from '../../../constants';

export default StyleSheet.create({
  commit: {
    borderBottomColor: primaryColor,
    borderBottomWidth: 1,
    alignContent: 'center',
    padding: 0,
    paddingHorizontal: 0,
    justifyContent: 'space-between',
    flex: 1,
  },
  text: {
    flexDirection: 'column',
    paddingTop: 8,
  },
  button: {
    margin: 12,
    marginHorizontal: 0,
    width: 100,
    backgroundColor: primaryColor,
    padding: 6,
    borderRadius: 8,
  },
  textButton: {
    color: 'white',
  },
});
