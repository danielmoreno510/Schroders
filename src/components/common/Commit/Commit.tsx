import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';

import styles from './styles';
import {formatDate} from '../../../utils/formUtils';

const Commit: React.FC<CommitProps> = ({commit, showCommit}) => (
  <View style={styles.commit}>
    <Text style={styles.text}>Author: {commit.user.login}</Text>
    <Text style={styles.text}>Date: {formatDate(commit.created_at)}</Text>
    <TouchableOpacity
      testID="showCommit"
      onPress={() => showCommit(commit)}
      style={styles.button}>
      <Text style={styles.textButton}>Go To Details</Text>
    </TouchableOpacity>
  </View>
);

export default Commit;
