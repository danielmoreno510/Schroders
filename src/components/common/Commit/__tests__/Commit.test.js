import React from 'react';
import TestRenderer from 'react-test-renderer';

import Commit from '../Commit';

describe('Commit component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      commit: {user: {login: ''}, created_at: new Date()},
      showCommit: jest.fn(),
    };

    describe('WHEN the component is rendered', () => {
      const testRenderer = TestRenderer.create(<Commit {...testProps} />);

      it('THEN should display a regular Commit', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });
  });
});
