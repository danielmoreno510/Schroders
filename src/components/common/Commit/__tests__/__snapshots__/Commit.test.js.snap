// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Commit 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "borderBottomColor": "#f1b30a",
      "borderBottomWidth": 1,
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 0,
      "paddingHorizontal": 0,
    }
  }
>
  <Text
    style={
      {
        "flexDirection": "column",
        "paddingTop": 8,
      }
    }
  >
    Author: 
  </Text>
  <Text
    style={
      {
        "flexDirection": "column",
        "paddingTop": 8,
      }
    }
  >
    Date: 
    2023-09-25
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#f1b30a",
        "borderRadius": 8,
        "margin": 12,
        "marginHorizontal": 0,
        "opacity": 1,
        "padding": 6,
        "width": 100,
      }
    }
    testID="showCommit"
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Go To Details
    </Text>
  </View>
</View>
`;
